tests/rag_answers/custom_deepeval/metrics/test_factual_correctness.py

```diff
@@ -0,0 +1,580 @@
+            assert isinstance(result, float)
+            assert result == metric.score
+
+        @pytest.mark.asyncio
+        async def test_non_native_model_empty_confusion_matrix_triggers_error(
+            self,
+            mock_non_native_model: Mock,
+            test_case,
+            caplog: pytest.LogCaptureFixture,
+        ):
+            mock_non_native_model.a_generate = AsyncMock(
+                return_value=FactClassificationResult(
+                    classified_facts=ClassifiedFacts()
+                )
+            )
+
+            caplog.set_level(logging.ERROR)
+            metric = FactualCorrectnessMetric(model=mock_non_native_model)
+
+            result = await metric.a_measure(test_case)
+
+            assert mock_non_native_model.a_generate.call_count == 1
+
+            captured_logs = caplog.text
+            assert (
+                "Error: no facts were classified. confusion_matrix is empty for input:"
+                in captured_logs
+            )
+
+            assert math.isnan(result)
+
+        @pytest.mark.parametrize(
+            "input_TP,input_FP,expected_score",
+            [
+                (
+                    ["fact1", "fact2"],
+                    ["fact3"],
+                    2 / 3,
+                ),
+                ([], [], 0.0),
+            ],
+        )
+        @pytest.mark.asyncio
+        async def test_non_native_model_return_scores(
+            self,
+            mock_non_native_model: Mock,
+            test_case: LLMTestCase,
+            input_TP: list,
+            input_FP: list,
+            expected_score: float,
+        ):
+            mock_non_native_model.a_generate = AsyncMock(
+                return_value=(
+                    FactClassificationResult(
+                        classified_facts=ClassifiedFacts(
+                            TP=input_TP, FP=input_FP, FN=["fact3"]
+                        )
+                    )
+                )
+            )
+
+            metric = FactualCorrectnessMetric(model=mock_non_native_model)  # type: ignore
+            result = await metric.a_measure(test_case)
+
+            assert round(result, 3) == round(expected_score, 3)
+
+            # assert dependency boundary was called once as expected (with schema)
+            mock_non_native_model.a_generate.assert_awaited_once()
+            _, kwargs = mock_non_native_model.a_generate.call_args
+            assert kwargs.get("schema") == FactClassificationResult
+
+        @pytest.mark.asyncio
+        async def test_non_native_model_does_not_set_evaluation_costs(
+            self,
+            mock_non_native_model: Mock,
+            test_case: LLMTestCase,
+        ):
+            metric = FactualCorrectnessMetric(model=mock_non_native_model)  # type: ignore
+            result = await metric.a_measure(test_case)
+
+            assert metric.evaluation_cost is None
+            assert isinstance(result, float)
+            assert result == metric.score
```

Comment:
```suggestion
```

I'd remove this as it's not really a factor in the scenario we're testing (and opens a question of why not assert on other instance variables e.g reason)